pkg/detectors/cloudflarecakey/cloudflarecakey.go

```
@@ -54,13 +54,19 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			req.Header.Add("Content-Type", "application/json")
 			req.Header.Add("user-agent", "curl/7.68.0") // pretend to be from curl so we do not wait 100+ seconds -> nice try did not work
 
-			req.Header.Add("X-Auth-User-Service-Key", resMatch)
-			res, err := client.Do(req)
+			req.Header.Add("X-Auth-User-Service-Key", caKey)
+			resp, err := client.Do(req)
 			if err == nil {
-				defer res.Body.Close()
-				if res.StatusCode >= 200 && res.StatusCode < 300 {
+				defer func() {
+					_, _ = io.Copy(io.Discard, resp.Body)
+					_ = resp.Body.Close()
+				}()
+
+				if resp.StatusCode == 200 {
```

This will silently ignore any non-200 responses.